src/pages/workspaces/workspace/LeaveWorkspaceModal.js

```diff
@@ -0,0 +1,50 @@
+import { useState } from 'react'
+import { div, h, span } from 'react-hyperscript-helpers'
+import { ButtonPrimary, spinnerOverlay } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
+import { reportError } from 'src/libs/error'
+import Events from 'src/libs/events'
+
+
+const LeaveWorkspaceModal = ({ workspace: { workspace: { workspaceId } }, onDismiss, onSuccess }) => {
+  const [leaving, setLeaving] = useState(false)
+  const helpText = 'Leave Workspace'
+
+  const leaveWorkspace = async () => {
+    try {
+      setLeaving(true)
+      await Ajax().Workspaces.leave(workspaceId)
+      Ajax().Metrics.captureEvent(Events.workspaceShare, { workspaceId })
+      setLeaving(false)
+      onDismiss()
+      onSuccess()
+    } catch (error) {
+      const { message } = await error.json()
+      Ajax().Metrics.captureEvent(Events.workspaceLeave, { workspaceId, errorMessage: message })
+      reportError(message)
+      setLeaving(false)
+      onDismiss()
+    }
+  }
+
+  return h(Modal, {
+    title: span({ style: { display: 'flex', alignItems: 'center' } }, [
+      icon('warning-standard', { size: 24, color: colors.warning() }),
+      span({ style: { marginLeft: '1ch' } }, [helpText])
+    ]),
+    styles: { modal: { background: colors.warning(0.1) } },
```

Comment:
The styles here look similar to the confirmation modal for deleting things. It may be possible to re-use that component. For example:

https://github.com/DataBiosphere/terra-ui/blob/4c71dba6556eeccb520dc2038202529f4fa0fae2/src/components/data/data-utils.js#L207-L230